squidDrone/tests/i2c/i2c_interface_tests.cpp

```diff
@@ -0,0 +1,139 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "i2c.hpp"
+
+namespace{
+
+  class I2CTests: public ::testing::Test{
+  protected:
+    virtual void SetUp(){
+      unit_under_test_ = std::make_unique<i2c::I2C>();
+    }
+
+    std::unique_ptr<i2c::I2C> unit_under_test_;
+
+    uint8_t address = 0x10;
+    uint16_t byte_size = 3;
+    uint32_t timeout = 2;
+    std::vector<uint8_t> data = {1, 2};
+
+    i2c::I2CStatus result_status;
+    std::vector<uint8_t> result_data;
+  };
+
+  TEST_F(I2CTests, read_successful){
+    std::tie(result_status, result_data) = unit_under_test_->Read(address, byte_size, timeout);
+
+    EXPECT_EQ(result_status, i2c::I2CStatus::I2C_TRANSACTION_SUCCESSFUL);
+    EXPECT_THAT(result_data, testing::ElementsAre(1, 2, 3));
+  }
+
+  TEST_F(I2CTests, read_failed){
+    address = 0x11;
+    std::tie(result_status, result_data) = unit_under_test_->Read(address, byte_size, timeout);
+
+    EXPECT_EQ(result_status, i2c::I2CStatus::I2C_TRANSACTION_FAILED);
+  }
+
+  TEST_F(I2CTests, read_timeout){
+    address = 0x12;
+    std::tie(result_status, result_data) = unit_under_test_->Read(address, byte_size, timeout);
+
+    EXPECT_EQ(result_status, i2c::I2CStatus::I2C_TRANSACTION_TIMEOUT);
+  }
+
+  TEST_F(I2CTests, read_busy){
+    address = 0x13;
+    std::tie(result_status, result_data) = unit_under_test_->Read(address, byte_size, timeout);
+
+    EXPECT_EQ(result_status, i2c::I2CStatus::I2C_TRANSACTION_BUSY);
+  }
+
+  TEST_F(I2CTests, read_byte_size_0){
```

Comment:
this is the lower illegal equivalency class, where is the upper illegal equivalency group (byte_size = 32+1)

Reply:
Added test in this commit: https://github.com/elheck/Squiddrone/pull/219/commits/e578c51813e066b04af12f4a910e87de8dfaf400